pkg/apis/ceph.rook.io/v1/types.go

```diff
@@ -2179,6 +2196,10 @@ type Device struct {
 	// +nullable
 	// +optional
 	Config map[string]string `json:"config,omitempty"`
+	// ResourceClassName is the unique name to fetch specific resourceLimts from ResourceClasses
+	// +nullable
+	// +optional
+	ResourceClassName string `json:"resourceClassName,omitempty"`
```

Comment:
I thought the request was to have the resource class on the node instead of the device as per https://github.com/rook/rook/issues/8752#issuecomment-925857505?

Reply:
Currently, we can add Resources directly at the node level, and all the devices should inherit that property.
So this is introduced to set different resources for different device sets of the node.

ResourceClass can be added to devices as well as nodes, and if the device doesn't set ResourceClass the node ResourceClass will be inherited.